packages/txservice/src/transaction.ts

```diff
@@ -0,0 +1,216 @@
+import { delay, jsonifyError } from "@connext/nxtp-utils";
+import { BigNumber, providers } from "ethers";
+import { BaseLogger } from "pino";
+
+import { TransactionServiceConfig } from "./config";
+import { ChainError } from "./error";
+import { ChainRpcProvider } from "./provider";
+import { FullTransaction, MinimalTransaction } from "./types";
+
+export class Transaction {
+
+  public responses: providers.TransactionResponse[] = [];
+  public receipt?: providers.TransactionReceipt;
+  public nonce?: number;
+  public nonceExpired = false;
+
+  // TODO: Not a fan of this arrangement, this code could be tightened up a bit.
+  public get gasPrice(): Promise<BigNumber> {
+    return this.getGasPrice();
+  }
+  private _gasPrice?: BigNumber;
+
+  public get fullTransaction(): Promise<FullTransaction> {
+    return new Promise(async (resolve) => 
+      resolve({
+        ...this.minTx,
+        gasPrice: await this.gasPrice,
+        nonce: this.nonce,
+      })
+    );
+  }
```

Comment:
Is there a chance of this completely swallowing up any errors around getting the gas price?